### ISSUE TITLE
redefined-var should warn inside top-level let bindings too

### ISSUE THREAD

**Issue by tomdl89** [clj-kondo/clj-kondo#1184]
**Is your feature request related to a problem? Please describe.**
Currently, redefined-var ignores redefinition inside let forms:

```
(let [foo 3]
  (def bar 4)
  (def bar 6)
```
will not be considered a linting error.

**Describe the solution you'd like**
Lint inside let too

**Describe alternatives you've considered**
Maybe generalise to lint inside all binding forms (binding, with-redefs, with-local-vars... etc?)

**Additional context**
@borkdude said:
>it you want to have a stab at this, it's in analyze-children in clj-kondo.impl.analyzer

[ Optional: if you are interested in doing a PR yourself, please leave a note. ] - I am

**Comment by rjerue** [comment]
Just a note that I did mess up the above impl, I _meant_ to say

```
(defn fun
  [{:keys [a-key]}]
  (when-let [{:keys [a-deep]} a-key])
  (println a-deep))
```

Still has the same error though
